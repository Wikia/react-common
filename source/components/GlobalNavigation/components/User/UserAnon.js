import React from 'react';
import { useTranslation } from 'react-i18next';
import PropTypes from 'prop-types';

import Dropdown from '../../../Dropdown';
import Avatar from '../../../Avatar';
import Button from '../../../Button';
import List from '../../../List';

import withRedirectUrl from '../../hocs/withRedirectUrl';

const UserToggle = ({ chevron }) => (
    <React.Fragment>
        <Avatar />
        {chevron}
    </React.Fragment>
);

UserToggle.propTypes = {
    chevron: PropTypes.node.isRequired,
};

const UserAnon = ({ data, getUrlWithRedirect }) => {
    const [t] = useTranslation();

    return (
        <Dropdown
            toggle={chevron => <UserToggle chevron={chevron} />}
            className="wds-global-navigation__user-menu wds-global-navigation__user-anon"
            contentScrollable={false}
            dropdownRightAligned
            hasShadow
            contentClassName="wds-global-navigation__dropdown-content"
        >
            <List linesBetween>
                <li>
                    <Button
                        fullwidth
                        href={getUrlWithRedirect(data.signin.href, data.signin['param-name'])}
                        rel="nofollow"
                        data-tracking-label={data.signin['tracking-label']}
                    >
                        {t(data.signin.title.key)}
                    </Button>
                </li>
                <li>
                    <div className="wds-global-navigation__user-menu-dropdown-caption">
                        Don&apos;t have an account?
                    </div>
                    <Button
                        href={getUrlWithRedirect(data.register.href, data.register['param-name'])}
                        rel="nofollow"
                        data-tracking-label={data.register['tracking-label']}
                        fullwidth
                        secondary
                    >
                        {t(data.register.title.key)}
                    </Button>
                </li>
            </List>
        </Dropdown>
    );
};

UserAnon.propTypes = {
    data: PropTypes.object.isRequired,
    getUrlWithRedirect: PropTypes.func.isRequired,
};

export default withRedirectUrl(UserAnon);
