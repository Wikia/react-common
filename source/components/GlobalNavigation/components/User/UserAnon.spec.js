import React from 'react';
import { shallow } from 'enzyme';
import merge from 'lodash/merge';

import UserAnon from './UserAnon';

const defaultProps = {
    data: {
        signin: {
            type: 'link-signin',
            href: 'https://www.wikia.com/signin',
            title: {
                type: 'translatable-text',
                key: 'global-navigation-user-sign-in',
            },
            'param-name': 'redirectto',
            'tracking-label': 'account.sign-in',
        },
        register: {
            type: 'link-register',
            href: 'https://www.wikia.com/register',
            title: {
                type: 'translatable-text',
                key: 'global-navigation-user-register',
            },
            'param-name': 'register-param',
            'tracking-label': 'account.register',
        },
    },
};

jest.mock('../../hocs/withRedirectUrl', () => Component => props => (
    <Component {...props} getUrlWithRedirect={(href, param) => `${href} + ${param}`} />
));

function renderComponent(props) {
    const computedProps = merge({}, defaultProps, props);

    return shallow(<UserAnon {...computedProps} />);
}

test('UserAnon renders correctly', () => {
    expect(renderComponent().dive()).toMatchSnapshot();
});

test('UserAnon renders toggle correctly', () => {
    const wrapper = renderComponent().dive();
    const Toggle = wrapper.find('Dropdown').props().toggle(<div />);

    expect(shallow(Toggle)).toMatchSnapshot();
});
