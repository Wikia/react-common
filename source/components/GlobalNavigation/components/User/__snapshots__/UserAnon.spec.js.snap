// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAnon renders correctly 1`] = `
<Dropdown
  canFlip={false}
  className="wds-global-navigation__user-menu wds-global-navigation__user-anon"
  contentClassName="wds-global-navigation__dropdown-content"
  contentScrollable={false}
  dropdownLeftAligned={false}
  dropdownRightAligned={true}
  hasDarkShadow={false}
  hasShadow={true}
  iconName="dropdown-tiny"
  isActive={false}
  isLevel2={false}
  isNotHoverable={false}
  isStickedToParent={false}
  noChevron={false}
  onClose={null}
  onMouseEnter={null}
  toggle={[Function]}
  toggleAttrs={Object {}}
  toggleClassName=""
>
  <List
    bigItems={false}
    boldItems={false}
    className=""
    hasEllipsis={false}
    isLinked={false}
    linesBetween={true}
  >
    <li>
      <Button
        className=""
        data-tracking-label="account.sign-in"
        disabled={false}
        fullwidth={true}
        href="https://www.wikia.com/signin + redirectto"
        light={false}
        onClick={[Function]}
        rel="nofollow"
        secondary={false}
        square={false}
        text={false}
      >
        global-navigation-user-sign-in
      </Button>
    </li>
    <li>
      <div
        className="wds-global-navigation__user-menu-dropdown-caption"
      >
        Don't have an account?
      </div>
      <Button
        className=""
        data-tracking-label="account.register"
        disabled={false}
        fullwidth={true}
        href="https://www.wikia.com/register + register-param"
        light={false}
        onClick={[Function]}
        rel="nofollow"
        secondary={true}
        square={false}
        text={false}
      >
        global-navigation-user-register
      </Button>
    </li>
  </List>
</Dropdown>
`;

exports[`UserAnon renders toggle correctly 1`] = `
<Fragment>
  <Avatar
    alt="User avatar"
  />
  <div />
</Fragment>
`;
